feature/contributor/src/main/java/io/github/droidkaigi/confsched2019/contributor/ContributorFragment.kt

```diff
@@ -0,0 +1,108 @@
+import com.xwray.groupie.GroupAdapter
+import com.xwray.groupie.Item
+import com.xwray.groupie.databinding.ViewHolder
+import dagger.Module
+import dagger.Provides
+import io.github.droidkaigi.confsched2019.contributor.action.ContributorActionCreator
+import io.github.droidkaigi.confsched2019.contributor.bindingmodel.ContributorBindingModel
+import io.github.droidkaigi.confsched2019.contributor.databinding.FragmentContributorBinding
+import io.github.droidkaigi.confsched2019.contributor.item.ContributorProfileItemHolder
+import io.github.droidkaigi.confsched2019.contributor.store.ContributorStore
+import io.github.droidkaigi.confsched2019.contributor.widget.DaggerFragment
+import io.github.droidkaigi.confsched2019.di.PageScope
+import io.github.droidkaigi.confsched2019.ext.android.changed
+import io.github.droidkaigi.confsched2019.system.actioncreator.ActivityActionCreator
+import io.github.droidkaigi.confsched2019.util.ProgressTimeLatch
+import me.tatarka.injectedvmprovider.InjectedViewModelProviders
+import javax.inject.Inject
+import javax.inject.Provider
+
+class ContributorFragment : DaggerFragment() {
+    private lateinit var binding: FragmentContributorBinding
+
+    @Inject
+    lateinit var contributorStoreProvider: Provider<ContributorStore>
+    private val contributorStore: ContributorStore by lazy {
+        InjectedViewModelProviders.of(requireActivity()).get(contributorStoreProvider)
+    }
+
+    @Inject
+    lateinit var contributorActionCreator: ContributorActionCreator
+    @Inject
+    lateinit var activityActionCreator: ActivityActionCreator
+
+    private val groupAdapter = GroupAdapter<ViewHolder<*>>()
+    private lateinit var progressTimeLatch: ProgressTimeLatch
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        binding = FragmentContributorBinding.inflate(
+            inflater,
+            container,
+            false
+        )
+        return binding.root
+    }
+
+    override fun onActivityCreated(savedInstanceState: Bundle?) {
+        super.onActivityCreated(savedInstanceState)
+
+        binding.contributorRecyclerView.layoutManager = LinearLayoutManager(requireContext())
+        binding.contributorRecyclerView.adapter = groupAdapter
+
+        progressTimeLatch = ProgressTimeLatch { showProgress ->
+            binding.progressBar.isVisible = showProgress
+        }.apply {
+            loading = true
+        }
+
+        contributorStore.loadingState.changed(viewLifecycleOwner) {
+            progressTimeLatch.loading = it.isLoading
+        }
+
+        contributorStore.contributorList.changed(viewLifecycleOwner) { result ->
+            val itemList = mutableListOf<Item<*>>()
+            itemList += result.contributors
+                .mapIndexed { index, item ->
+                    ContributorBindingModel(index, item.name, item.iconUrl, item.profileUrl)
```

Comment:
not only name field, because Contributor instance not have index field.
index field -> for show ContributorRanking(?). Not use fun bind(binding,index) because, correct index.
iconUrl field -> for show contributor icon image
profileUrl -> for open Profile Page

Reply:
Anyway, we don't use presentation model pattern for now. Because It is because we do not use that method at this project.
I think If you can write a lot of logic in the presentation model it may be worth it. The current situation is not a problem by writing logic to an ordinary model.

Can you write it like this?

```kotlin
result.contributors
                .mapIndexed { index, contributer ->
                        ContributorItem(contributer, index)
               }
```